kview/src/main/java/dev/ikm/komet/kview/mvvm/view/details/DetailsController.java

```diff
@@ -254,6 +316,45 @@ public DetailsController(UUID conceptTopic) {
 
     @FXML
     public void initialize() {
+        AtomicInteger count = new AtomicInteger();
+        identiconImageView.setOnContextMenuRequested(contextMenuEvent -> {
+            // query all available memberships (semantics having the purpose as 'membership', and no fields)
+            // query current concept's membership semantic records.
+            // build menuItems according to 'add' or 'remove' , style to look like figma designs style classes.
+            // show offset to the right of the identicon
+            ViewCalculator viewCalculator = viewProperties.calculator();
+            EntityFacade currentConceptFacade = conceptViewModel.getPropertyValue(CURRENT_ENTITY);
+            List<PatternEntityVersion> patterns = getMembershipPatterns();
+            ContextMenu membershipContextMenu = new ContextMenu();
+            membershipContextMenu.getStyleClass().add("kview-context-menu");
+
+            List<MenuItem> addedMenuItems = new ArrayList<>();
+            List<MenuItem> removedMenuItems = new ArrayList<>();
+            for (PatternEntityVersion pattern : patterns) {
+                MenuItem menuItem = new MenuItem();
+                if (isInMembershipPattern(currentConceptFacade.nid(), pattern.nid())) {
+                    menuItem.setText("Remove from " + pattern.entity().description());
+                    menuItem.setOnAction(evt -> removeFromMembershipPattern(currentConceptFacade, pattern.entity(), viewCalculator));
+                    addedMenuItems.add(menuItem);
+                } else {
+                    menuItem.setText("Add to " + pattern.entity().description());
+                    menuItem.setOnAction(evt -> addToMembershipPattern(currentConceptFacade, pattern.entity(), viewCalculator));
+                    removedMenuItems.add(menuItem);
+                }
+                addPlusIconToMenuItem(menuItem);
+            }
+            // sort the added (able to be removed)
+            Comparator<MenuItem> patternMenuComparator = (m1, m2) -> m1.getText().compareToIgnoreCase(m2.getText());
+            membershipContextMenu.getItems().addAll(addedMenuItems.stream().sorted(patternMenuComparator).collect(Collectors.toList()));
+            // then add a menu line separator
+            membershipContextMenu.getItems().add(new SeparatorMenuItem());
```

Comment:
Can we check if there are any remove entries before adding the the separator menu item?

Reply:
done